domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/ApplicationLifecycleObserver.kt

```diff
@@ -71,4 +86,49 @@ class ApplicationLifecycleObserver @Inject constructor(
       }
     }
   }
+
+  private fun logRelativeCpuUsageInBackground() {
+    CoroutineScope(backgroundDispatcher).launch {
+      while (true) {
+        val previousCpuUsageParameters = performanceMetricsController.getLastCpuUsageParameters()
+        val currentCpuUsageParameters = performanceMetricsController.getCurrentCpuUsageParameters(
+          currentApplicationState
+        )
+        val relativeCpuUsage = performanceMetricsController.getRelativeCpuUsage(
+          previousCpuUsageParameters, currentCpuUsageParameters
+        )
+        val applicationState = calculateRelativeApplicationState(
+          previousCpuUsageParameters.applicationState,
+          currentApplicationState
+        )
+        val currentScreen = activityLifecycleObserver.getCurrentScreen()
+        performanceMetricsLogger.logCpuUsage(currentScreen, applicationState, relativeCpuUsage)
+        performanceMetricsController.cacheCpuUsageParameters(currentCpuUsageParameters)
+        delay(timeIntervalInMillis)
```

Comment:
FWIW I have a feeling this this whole logging routine will need to be gated by a platform parameter flag since the loop is likely to cause an livelock when using advanceUntilIdle in TestCoroutineDispatchers.

This will probably be more of an issue for our Espresso IdlingResource, though.

Reply:
I've used the already available `enablePerformanceMetricsLogging` platform parameter for this -- does that work?

Reply:
Yep that sounds good.

Reply:
I think this flag isn't required now, so I've removed it. Or, is there a place you'd still want me to put it?

Reply:
Hmm. Shouldn't the new snapshotter be gated? While you've add gating logic for the event logger itself, the new run-coroutine-always behavior of the snapshotter class will probably result in a livelock with advanceUntilIdle (so I think the gate is necessary at a higher level specifically for CPU performance collection).

Reply:
cool -- gated snapshotter at applicationLifecycleObserver.

Reply:
Thanks! Can you please make sure there are also tests to verify that the logging is properly disabled when the flag is off? Given that this is a long-term behavior we'll probably rely on, it'd be good to be sure that it works as expected (and continues to do so).

Reply:
added additional tests for testing default behaviour of both applicationLifecycleObserver and PerformanceMetricsLogger.